storm-client/src/jvm/org/apache/storm/Config.java

```diff
@@ -1048,6 +1048,14 @@
     @NotNull
     public static final String SUPERVISOR_WORKER_TIMEOUT_SECS = "supervisor.worker.timeout.secs";
     /**
+     * Topology configurable worker heartbeat timeout before the supervisor tries to restart the worker process.
+     * Overrides {@link #SUPERVISOR_WORKER_TIMEOUT_SECS}
+     */
+    @IsInteger
+    @IsPositiveNumber
+    @NotNull
+    public static final String TOPOLOGY_WORKER_TIMEOUT_SECS = "topology.worker.timeout.secs";
```

Comment:
Because the behavior can be different, it would be good be document what will happen if this is used in terms of 

`supervisor.worker.timeout.secs`, `supervisor.worker.start.timeout.secs`, `nimbus.task.timeout.secs`,  and `nimbus.task.launch.secs`.